compiler/src/main/java/com/xfinity/resourceprovider/RpCodeGenerator.java

```diff
@@ -268,6 +295,45 @@ TypeSpec generateIntegerProviderClass() {
         return classBuilder.build();
     }
 
+    TypeSpec generateIdProviderClass() {
+        MethodSpec constructor = MethodSpec.constructorBuilder()
+                                           .addModifiers(PUBLIC)
+                                           .addParameter(contextClassName, "context")
+                                           .addStatement("this.context = context")
+                                           .build();
+
+        TypeSpec.Builder classBuilder = classBuilder("IdProvider")
```

Comment:
The indents are messed up here - because classBuilder doesn't make an immediate chained call on the same line, the subsequent calls are aligned to the . in the TypeSpec.Builder, not to the related calls.